### PR TITLE
Avoidance Feedback Rebased

### PR DESCRIPTION
**Pull request by mrivi**
Based on #10037 

- if the vehicle projected position on the line previous-current triplet is past the current triplet, acceptance radius is enlarged such that navigator updates the triplets
- if the vehicle is within the xy acceptance radius but not inside the z one, enlarge the altitude acceptance radius such that navigator updates the triplets set the feedback yaw acceptance to NAN such that it is ignored by navigator

Missions shouldn't get stuck anymore because a waypoint is not reachable or a waypoint is reached with a heading different from what navigator expects

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@mrivi Could you please rebase one more time and check CI? Flash space is now resolved, so any remaining failures should be real.

File: src/modules/gnd_pos_control/GroundRoverPositionControl.cpp [hunk 1]
```diff
@@ -447,22 +444,22 @@ GroundRoverPositionControl::task_main()
...
-
-					_gnd_pos_ctrl_status.target_bearing = _gnd_control.target_bearing();
-					_gnd_pos_ctrl_status.xtrack_error = _gnd_control.crosstrack_error();
-
-					matrix::Vector2f curr_wp((float)_pos_sp_triplet.current.lat, (float)_pos_sp_triplet.current.lon);
-					_gnd_pos_ctrl_status.wp_dist = get_distance_to_next_waypoint((double)current_position(0), (double)current_position(1),
-								       (double)curr_wp(0), (double)curr_wp(1));
+					_gnd_pos_ctrl_status.acceptance_radius = turn_distance;
+
+//					_gnd_pos_ctrl_status.nav_roll = _gnd_control.get_roll_setpoint();
```
Comment (LorenzMeier) [inline src/modules/gnd_pos_control/GroundRoverPositionControl.cpp]: Why is this code here just commented out?

Comment (mrivi) [reply src/modules/gnd_pos_control/GroundRoverPositionControl.cpp]: @dagar how can we bring this in? Could we use this new message only for avoidance and then in a new PR use it for VTOL as well?

Comment (LorenzMeier) [reply src/modules/gnd_pos_control/GroundRoverPositionControl.cpp]: @mrivi You are not filling the payload of the message but still publishing it below - is this intended?

Comment (dagar) [reply src/modules/gnd_pos_control/GroundRoverPositionControl.cpp]: This was from the original PR where fw_pos_ctrl_status was being generalized. Some of these fields are straight copies of what also goes out in vehicle_attitude_setpoint, so I was going to drop them.
https://github.com/PX4/Firmware/pull/10037

Let's finish https://github.com/PX4/Firmware/pull/10037 and rebase.

Comment (mrivi) [reply src/modules/gnd_pos_control/GroundRoverPositionControl.cpp]: @dagar I have rebased.

**Pull request closed without merging**